crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -49,6 +49,30 @@ impl Settlement {
         }
 
         let auction = persistence.get_auction(solution.auction_id()).await?;
+        let (winner, winner_score, deadline) =
+            persistence.get_winner(solution.auction_id()).await?;
+
+        if transaction.solver != winner {
+            return Err(Error::WinnerMismatch {
+                expected: winner,
+                got: transaction.solver,
+            });
+        }
+
+        let score = solution.score(&auction)?;
+        if score != winner_score {
+            return Err(Error::ScoreMismatch {
+                expected: winner_score,
+                got: score,
+            });
+        }
+
+        if transaction.block >= deadline {
+            tracing::warn!(
+                "Settlement for auction {} was submitted after the deadline",
+                solution.auction_id()
+            );
+        }
```

Comment:
Do we want to allow this? Do we want to keep some "leeway" but not have unlimited delay?

Reply:
For OnSettlementEventUpdater (right now, since this is a refactor for current behaviour) we want to allow this and save observation to database. For circuit breaker I am not sure what is a desired behaviour, so will have to consult the solver team and how they handle it.